js/modules/k6/http/request.go

```diff
@@ -89,9 +89,20 @@ func (h *HTTP) Options(ctx context.Context, url goja.Value, args ...goja.Value)
 // Request makes an http request of the provided `method` and returns a corresponding response by
 // taking goja.Values as arguments
 func (h *HTTP) Request(ctx context.Context, method string, url goja.Value, args ...goja.Value) (*Response, error) {
+	state := lib.GetState(ctx)
+	var throw bool
+	if state != nil {
+		throw = state.Options.Throw.Bool
+	}
```

Comment:
`state` should be non-`nil` here, though unfortunately that is checked only in `parseRequest()`, which is called further down below... 
https://github.com/loadimpact/k6/blob/a379c3787372ce2a4c70cb2049fc43be7af798a2/js/modules/k6/http/request.go#L107
https://github.com/loadimpact/k6/blob/a379c3787372ce2a4c70cb2049fc43be7af798a2/js/modules/k6/http/request.go#L121-L128

Given that we must implement the same `throw` check in `http.batch()` (i.e. fix the similar code [here](https://github.com/loadimpact/k6/blob/a379c3787372ce2a4c70cb2049fc43be7af798a2/js/modules/k6/http/request.go#L476-L479) and [here](https://github.com/loadimpact/k6/blob/a379c3787372ce2a4c70cb2049fc43be7af798a2/js/modules/k6/http/request.go#L491-L493) :disappointed: ), maybe the best way would be delaying the parsing of the URL and doing it in `parseRequest()`, so we only change one place?

Reply:
I am for even moving further and in the `httpext` package:
1. otherwise error codes will need to become exported (and probably moved)
2. we already need to then stringify the URL back ... which is wasteful, given that we already have it as a string

This might have other implications as I haven't gone around the code to look for places where this will not be easier

Reply:
I think https://github.com/loadimpact/k6/blob/a379c3787372ce2a4c70cb2049fc43be7af798a2/js/modules/k6/http/http_url.go#L49 will make moving all of the URL parsing in [httpext.MakeRequest](https://github.com/loadimpact/k6/blob/a379c3787372ce2a4c70cb2049fc43be7af798a2/lib/netext/httpext/request.go#L174) ... impossible :(

Reply:
Since both, `Request` and `Batch` call `parseRequest()` I agree with @na--'s suggestion on moving the URL validation there. we could replace `reqURL httpext.URL` by `reqURL gojaValue`.